### ISSUE TITLE
Possible changes in how Home displays primary accessory

### ISSUE THREAD

**Issue by imanel** [HomeSpan/HomeSpan#244]
First things first - amazing project, thanks for your work!

I'm currently going through examples and learning along the way, and I'm at example 11 (ServiceOptions). In this example, there are multiple cases shown on how to change the primary device, however, in my Home app, all Ceiling Fans except #5 are visible as light. That would suggest that `setPrimary` stopped working for some reason. What is also interesting is it looks like `Ceiling Fan #3` is working now correctly (shows as `Ceiling Fan #3` instead of just `Fan`), but still is represented by light.

Is this the issue only with my configuration, or has Apple changed something in the recent version? Tested on both iPhone and Mac, both show the same result. I also unpaired and paired again everything, still had the same result.

HomeSpan 1.4.2
iOS 15.3
macOS 12.2

**Comment by HomeSpan** [comment]
Thanks for noting this.  Indeed Apple seems to have changed things and the Home App no longer honors setPrimary() the way it previously did when the examples were first created.  In some cases I found the Home App picks the first Service listed as the primary, but even that is not always the case.

There is also another related issue - The Name() Characteristic is not honored for all Services.  Early in iOS 14 the name shown on an Accessory Tile would either match the name provided in the Information Service, or it would match the Name() Characteristic of a specific Service if provided.  That no longer seems to be the case.  Even worse, sometimes the Name() Characteristic provided in the Information Service is completely ignored, and the name shown on the Accessory Tile is inherited from the MDNS name broadcast for pairing (this is the case with the TV Service).

Icons on tiles are similarly inconsistent.

I think the root of the problem is that Apple has not updated HAP-R2 even though they have clearly made changes and enhancements to HomeKit for commercial users.  The TV Service is one good example of an enhancement that is not documented anywhere by Apple but clearly exists.  But to make the TV Service work Apple had to make changes to Characteristic Name(), which is ignored altogether, and has been replaced with a new (undocumented) Characteristic ConfiguredName().

Fortunately, most of these issues are purely cosmetic - the core functionality of the Tiles still work as expected.  But it is disappointing that Apple has made updates in ways that are difficult to keep track of.

I may modify Example 11, or at least add a caveat to the documentation, indicating that it no longer tracks the Home App as expected.
